feature_engine/selection/drop_features.py

```diff
@@ -52,12 +52,15 @@ class DropFeatures(BaseSelector):
 
     def __init__(self, features_to_drop: List[Union[str, int]]):
 
-        if not isinstance(features_to_drop, list) or len(features_to_drop) == 0:
+        if not isinstance(features_to_drop, (str, list)) or len(features_to_drop) == 0:
             raise ValueError(
-                "features_to_drop should be a list with the name of the variables"
+                "features_to_drop should be a list with the name of the variables "
                 "you wish to drop from the dataframe."
             )
 
+        if isinstance(features_to_drop, str):
```

Comment:
To follow sklearn conventions, we must not re-assign user' entered values in the init parameters. 

Here, we need to let feature_to_drop be a string, and make it a list when we create features_to_drop_ in the fit method.

I am surprised that the `check_estimator` from sklearn did not pick this up. I was expecting the test to fail :_(